### PR TITLE
[bugfix] logql: vectorExpr run fail when query_range: parallelise_shardable_queries: true

### PR DESCRIPTION
**Pull request by liguozhong**
…hardable_queries: true

**What this PR does / why we need it**:
A grafana error occurred when I deployed the vector(0) function to our online  loki cluster.
It was finally found because online loki has a configuration

`unexpected expr type (*syntax.VectorExpr) for ASTMapper type (logql.ShardMapper)
`
```yaml
query_range:
   parallelise_shardable_queries: true
```
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
error snapssnapshot
image
with this pr，success snapshot
image

**Checklist**
- [ ] Documentation added
- [ ] Tests updated
- [ ] Is this an important fix or new feature? Add an entry in the `CHANGELOG.md`.
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`

### CODE REVIEW

**Review by kavirajk — COMMENTED** [review]
Thanks @liguozhong. Should we add some tests?

**Comment by liguozhong** [comment]
> Thanks @liguozhong. Should we add some tests?

👌

**Comment by liguozhong** [comment]
> Thanks @liguozhong. Should we add some tests?

done,thanks.

File: pkg/logql/shardmapper_test.go [hunk 1]
```diff
@@ -782,6 +782,67 @@ func TestMapping(t *testing.T) {
 				},
 			},
 		},
+		{
+			in: `vector(0) or sum (rate({foo="bar"}[5m]))`,
```
Comment (kavirajk) [inline pkg/logql/shardmapper_test.go:786]: This test seems to be failing.
https://drone.grafana.net/grafana/loki/15228/2/5

Comment (liguozhong) [reply pkg/logql/shardmapper_test.go:786]: The original testcase is indeed wrong, and the 'Group' field is not considered.
 I just submitted a commit to fix this testcase

**Review by kavirajk — APPROVED** [review]
thanks @liguozhong LGTM

**Pull request merged**